### PR TITLE
Use oauth2 clients for all auth methods

### PR DESCRIPTION
**Pull request by dirkkul**
Use authilb.Oauth2 sessions to manage authentication instead of adding a authorization header manually.

This has multiple advantages:
- Less requests where a body has to be created and a response parsed
- Oauth2 sessions acts like a request session while handling all the authentication in the backround
- You can refresh tokens using session.refresh_token(). So it is not needed to keep username and password in memory.

While oauth2 automatically refreshes tokens, it does not have an internal wake-up timer. In case the client idles longer than the lifetime of the refresh token, the authorization can time out. To circumvent this, I added a backround task that periodically refreshes the tokens.

Wit this change, I gave priority to the `auth_client_secret` parameter over the authentication header. As it is not possible to add refresh tokens there, `auth_client_secret` is better in all cases. 

I also had to delete some unit tests. They were testing the old implementation and there was no real way to transfer them over. I think long-term we need to replace these tests with tests that use a mock-server.

### CODE REVIEW

File: ci/docker-compose-okta-cc.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 version: '3.4'
 services:
-  weaviate-auth-okta:
+  weaviate-auth-okta-cc:
```
Comment (parkerduckworth) [inline ci/docker-compose-okta-cc.yml:4]: what is the meaning of `cc` here?

File: weaviate/connect/connection.py [hunk 2]
```diff
@@ -67,94 +73,115 @@ def __init__(
...
+                    new_session = _auth.get_auth_session()
+                    self._session.token = new_session.fetch_token()
+
+        self._background_thread = Thread(
+            target=periodic_refresh_token,
+            args=(expires_in, _auth),
+            daemon=True,
+            name="TokenRefresh",
+        )
+        self._background_thread.start()
```
Comment (parkerduckworth) [inline weaviate/connect/connection.py:176]: is this thread automatically cleaned up in case of an ungraceful client crash?

**Review by parkerduckworth — APPROVED** [review]

File: ci/docker-compose-okta-cc.yml (hunk 1 above)
Comment (dirkkul) [reply ci/docker-compose-okta-cc.yml:4]: client credentials

File: weaviate/connect/connection.py (hunk 2 above)
Comment (dirkkul) [reply weaviate/connect/connection.py]: good point, done

File: weaviate/warnings.py [hunk 3]
```diff
@@ -13,6 +14,47 @@ def auth_with_anon_weaviate():
...
 
+    @staticmethod
+    def auth_no_refresh_token(auth_len: Optional[int] = None):
+        if auth_len is not None:
+            msg = f"The current access token is only valid for {auth_len}s."
+        else:
+            msg = "Also, no expiration time was given."
+
+        warnings.warn(
+            message=f"""Auth002: The token returned from you identity provider does not contain a refresh token. {msg}
```
[the remark this answers is not in the document]
Comment (parkerduckworth) [reply weaviate/warnings.py:25]: This is ok with me, your explanation makes total sense :)

**Pull request merged**